client/src/variants/Remarkablerookies.js

```diff
@@ -0,0 +1,205 @@
+import { ChessRules } from "@/base_rules";
+
+export class RemarkablerookiesRules extends ChessRules {
+
+  static get PawnSpecs() {
+    return Object.assign(
+      {},
+      ChessRules.PawnSpecs,
+      {
+        promotions:
+          ChessRules.PawnSpecs.promotions.concat(
+          [V.SHORT_ROOK, V.WOODY_ROOK, V.HALF_DUCK, V.CHANCELLOR])
+      }
+    );
+  }
+
+  getPpath(b) {
+    if ([V.SHORT_ROOK, V.WOODY_ROOK, V.HALF_DUCK, V.CHANCELLOR].includes(b[1]))
+      return "Remarkablerookies/" + b;
+    return b;
+  }
+
+  static GenRandInitFen(randomness) {
+    if (randomness == 0)
+      return "dhaskahd/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w 0 ahah -";
+
+    // Mapping white --> black (at least at start):
+    const piecesMap = {
+      'r': 'd',
+      'n': 'h',
+      'b': 'a',
+      'q': 's',
+      'k': 'k'
+    };
+
+    const baseFen = ChessRules.GenRandInitFen(randomness);
+    return (
+      baseFen.substr(0, 8).split('').map(p => piecesMap[p]).join('') +
+      baseFen.substr(8)
+    );
+  }
+
+  static get SHORT_ROOK() {
+    return 'd';
+  }
+  static get WOODY_ROOK() {
+    return 'h';
+  }
+  static get HALF_DUCK() {
+    return 'a';
+  }
+  static get CHANCELLOR() {
+    return 's';
+  }
+
+  static get PIECES() {
+    return (
+      ChessRules.PIECES.concat([V.SHORT_ROOK, V.WOODY_ROOK, V.HALF_DUCK, V.CHANCELLOR])
+    );
+  }
+
+  getPotentialMovesFrom([x, y]) {
+    switch (this.getPiece(x, y)) {
+      case V.SHORT_ROOK: return this.getPotentialShortRookMoves([x, y]);
+      case V.WOODY_ROOK: return this.getPotentialWoodyRookMoves([x, y]);
+      case V.HALF_DUCK: return this.getPotentialHalfDuckMoves([x, y]);
+      case V.CHANCELLOR: return this.getPotentialChancellorMoves([x, y]);
+      default: return super.getPotentialMovesFrom([x, y]);
+    }
+    return [];
+  }
+
+  static get steps() {
+    return Object.assign(
+      {},
+      ChessRules.steps,
+      {
+        // "$" prefix to avoid conflict/confusion with piece abbreviations
```

Comment:
Not sure if you'd prefer I just used 'd' 'w' etc instead of '$d' '$w' etc -- can change of course